src/compat/posix/time/strptime.c

```diff
@@ -0,0 +1,327 @@
+		return dec;
+	}
+	while (count != num) {
+		res *= dec;
+		count++;
+	}
+	return res;
+}
+
+int transform_to_int(char *new_s, int num) {
+	int count = 0;
+	int a, b = 0;
+	int c;
+	while ((new_s[count] > 47) && (new_s[count] < 58)) {
+		c = degree(10, num-1);
+		a = (new_s[count] - '0')*c;
+		b+=a;
+		count++;
+		num--;
+	}
+	return b;
+}
+
+char *cmp_before_procent(char *new_s, char *new_form) {
+	int count = 0;
+	while (new_form[count] != '\0') {
+		if (new_form[count] == '%') {
+			return &new_form[count];
+		}
+		if ((new_form[count] > 64) && (new_form[count] < 91)) {
+			new_form[count] = tolower(new_form[count]);
+		}
+		if (strncmp(&new_form[count], &new_s[count], 1) != 0) {
+			return NULL;
+		}
+		count++;
+	}
+	return &new_form[count];
+}
+
+char *def_format(char *new_form, char *new_s, struct tm *tm) {
+
+	char arr_day[7][10] = {"sunday", "monday", "tuesday", "wednesday", "thursday", "friday", "saturday"};
+	char arr_month[6][10] = {"january", "february", "march", "april", "may", "june"};
+	char arr_month_2[6][10] = {"july", "august", "september", "october", "november", "december"};
+	int n_leng = num_length(&new_s[0]);
+	int w_leng = word_length(&new_s[0]);
+	int date;
+	int str_day_leng[6] = { 3, 6, 7, 8, 9, '\0'};
+	int str_month_leng[8] = { 3, 4, 5, 6, 7, 8, 9, '\0'};
+	int count_jan_jun = 0;
+	int count_jul_dec = 0;
+	int count_wday = 0;
+	int count_leng = 0;
+
+	switch (*new_form) {
+		case 'm':
+		case 'w':
+		case 'y':
+		case 'Y':
+		case 'H':
+		case 'I':
+		case 'M':
+		case 'S':
+		case 'j':
+		case 'd':
+		case 'e':
+		case 'C':
+		case 'U':
+		case 'W':
+			if (n_leng == 0) {
+				return NULL;
+			}
+			date = transform_to_int(new_s, n_leng);
+			break;
+	}
+
+	switch (*new_form) {
+		case 'a':
+		case 'A':
+			if (w_leng == 0) {
+				return NULL;
+			}
+			while (str_day_leng[count_leng] != '\0') {
+				if (str_day_leng[count_leng] != w_leng) {
+					count_leng++;
+				} else {
+					while (count_wday != 7) {
+						if (strncmp(new_s, arr_day[count_wday], w_leng) == 0) {
+							tm->tm_wday = count_wday;
+							return &new_s[w_leng];
+						}
+						count_wday++;
+					}
+				}
+			}
+			return NULL;
+
+		case 'b':
+		case 'B':
+		case 'h':
+			if (w_leng == 0) {
+				return NULL;
+			}
+			while (str_month_leng[count_leng] != '\0') {
+				if (str_month_leng[count_leng] != w_leng) {
+					count_leng++;
+				} else {
+					while (count_jan_jun != 6) {
+						if (strncmp(new_s, arr_month[count_jan_jun], w_leng) == 0) {
+							tm->tm_mon = count_jan_jun;
+							return &new_s[w_leng];
+						}
+						count_jan_jun++;
+					}
+					while (count_jul_dec != 6) {
+						if(strncmp(new_s, arr_month_2[count_jul_dec], w_leng) == 0) {
+							tm->tm_mon = count_jan_jun+count_jul_dec;
+							return &new_s[w_leng];
+						}
+						count_jul_dec++;
+					}
+					return NULL;
+				}
+			}
+			return NULL;
+
+		case 'm':
+			if ((n_leng > 2) || (date > 12) || (date == 0)) {
+				return NULL;
+			}
+			tm->tm_mon = date-1;
```

Comment:
Add spaces around the `-` operator to fit code style.

Reply:
I fixed it. Thank you!